kratos/utilities/properties_accessor.h

```diff
@@ -0,0 +1,322 @@
+//    |  /           |
+//    ' /   __| _` | __|  _ \   __|
+//    . \  |   (   | |   (   |\__ `
+//   _|\_\_|  \__,_|\__|\___/ ____/
+//                   Multi-Physics
+//
+//  License:		 BSD License
+//					 Kratos default license: kratos/license.txt
+//
+//  Main authors:    Philipp Bucher
+//
+
+#if !defined(KRATOS_PROPERTIES_ACCESSOR_H)
+#define  KRATOS_PROPERTIES_ACCESSOR_H
+
+// System includes
+#include <map>
+
+// External includes
+
+// Project includes
+#include "includes/define.h"
+#include "containers/data_value_container.h"
+#include "geometries/geometry.h"
+#include "includes/node.h"
+#include "includes/process_info.h"
+#include "includes/properties.h"
+
+
+namespace Kratos
+{
+///@addtogroup KratosCore
+///@{
+
+///@name Kratos Globals
+///@{
+
+///@}
+///@name Type Definitions
+///@{
+
+///@}
+///@name  Enum's
+///@{
+
+///@}
+///@name  Functions
+///@{
+
+///@}
+///@name Kratos Classes
+///@{
+
+/// Short class definition.
+/** Detail class definition.
+*/
+class PropertiesAccessor
+{
+public:
+    ///@name Type Definitions
+    ///@{
+
+    /// Pointer definition of PropertiesAccessor
+    KRATOS_CLASS_POINTER_DEFINITION(PropertiesAccessor);
+
+    using NodeType = Node<3>;
+    using GeometryType = Geometry<NodeType>;
+
+    using IndexType = std::size_t;
+    using SizeType = std::size_t;
+
+    using DoubleVariableType = Variable<double>;
+    using ComponentVariableType = VariableComponent< VectorComponentAdaptor<array_1d<double, 3> > >;
+    using Array3VariableType = Variable<array_1d<double, 3>>;
+
+    using DoubleFunction = std::function<double(const Properties*,
```

Comment:
I would pass a Properties& instead (actually const Properties& is EXACTLY Properties& ... but just for semantics i think the former is better since it was invented for this)

Reply:
done